scrapy/spiders/__init__.py

```diff
@@ -72,7 +78,54 @@ def _set_crawler(self, crawler: Crawler) -> None:
         self.settings: BaseSettings = crawler.settings
         crawler.signals.connect(self.close, signals.spider_closed)
 
-    def start_requests(self) -> Iterable[Request]:
+    async def yield_seeds(self) -> AsyncIterator[Any]:
```

Comment:
Is it intentional that this is `AsyncIterator` and not just `AsyncIterable`? If we only use it with `async for` the latter is enough.

Reply:
I am unfamiliar with the differences between the 2, you are probably right.

Reply:
*Iterator is a superset of *Iterable, with the additions that are usually not needed. That's basically all I know without checking docs every time.